Makefile

```diff
@@ -46,6 +47,17 @@ lint: $(PY_SPEC_ALL_TARGETS)
 	cd $(PY_SPEC_DIR); . venv/bin/activate; \
 	flake8 --max-line-length=120 ./eth2spec;
 
+install_deposit_contract_test:
+	cd $(PY_SPEC_DIR); python3 -m venv venv; . venv/bin/activate; \
+	cd ../..; cd $(DEPOSIT_CONTRACT_DIR); \
+	pip3 install -r requirements-testing.txt
+
+test_deposit_contract: $(PY_SPEC_ALL_TARGETS)
+	cd $(PY_SPEC_DIR); python3 -m venv venv; . venv/bin/activate; \
+	cd ../.. && cd $(DEPOSIT_CONTRACT_DIR); \
+	pip3 install -r requirements-testing.txt; \
```

Comment:
Is this necessary? Is the intention to make it fail safe?

Right now, `install_test` and `test` are independent. Maybe we should ask @protolambda his logic here.

Reply:
As I mentioned in the PR description, if we remove this line, the CI would throw:
```sh
==================================== ERRORS ====================================
_______________ ERROR collecting tests/contracts/test_deposit.py _______________
ImportError while importing test module '/home/circleci/specs-repo/deposit_contract/tests/contracts/test_deposit.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
tests/contracts/test_deposit.py:20: in <module>
    from eth2spec.phase0.spec import (
E   ModuleNotFoundError: No module named 'eth2spec.phase0.spec'
!!!!!!!!!!!!!!!!!!! Interrupted: 1 errors during collection !!!!!!!!!!!!!!!!!!!!
=========================== 1 error in 1.15 seconds ============================
Makefile:56: recipe for target 'test_deposit_contract' failed
make: *** [test_deposit_contract] Error 2
Exited with code 2
```

And if we added it, when executing this line, basically most packages will be "Requirement already satisfied" instead of `pyspec` (see: https://circleci.com/gh/ethereum/eth2.0-specs/2681?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link).

I guess between the different CircleCI jobs, the local requirement would be lost. Right now having this line doesn't increase too much overhead since all other packages are detected to "installed".

Reply:
> Right now, install_test and test are independent.

People run tests locally repeatedly, but testing requirements only have to be installed once, running those over again (even if cached) just takes extra time, but that slowdown is nice to avoid.

Reply:
I finally figured out how to fix it: https://github.com/ethereum/eth2.0-specs/pull/1127/commits/e52c4a5526866deba675ef11ac6c4cf8499dba65 😅